jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/IdleTimeoutTest.java

```diff
@@ -686,51 +695,57 @@ public boolean process(Request request, Response response, Callback callback)
+                ByteBuffer data = ByteBuffer.allocate(10);
+                stream.data(new DataFrame(stream.getId(), data, true), Callback.NOOP);
+
+                if (!phaser.get().await(1, TimeUnit.SECONDS))
+                    break;
+            }
+
+            // Send one more request to consume the whole session flow control window.
+            CountDownLatch resetLatch = new CountDownLatch(1);
+            MetaData.Request request = newRequest("GET", HttpFields.EMPTY);
             HeadersFrame frame = new HeadersFrame(request, null, false);
             FuturePromise<Stream> promise = new FuturePromise<>();
-            client.newStream(frame, promise, null);
+            client.newStream(frame, promise, new Stream.Listener()
+            {
+                @Override
+                public void onReset(Stream stream, ResetFrame frame, Callback callback)
+                {
+                    callback.succeeded();
+                    resetLatch.countDown();
+                }
+            });
             Stream stream = promise.get(5, TimeUnit.SECONDS);
-            ByteBuffer data = ByteBuffer.allocate(10);
+            ByteBuffer data = ByteBuffer.allocate(((HTTP2Session)client).updateSendWindow(0));
             stream.data(new DataFrame(stream.getId(), data, true), Callback.NOOP);
 
-            if (!phaser.get().await(1, TimeUnit.SECONDS))
-                break;
-        }
+            assertTrue(resetLatch.await(2 * idleTimeout, TimeUnit.MILLISECONDS));
 
-        // Send one more request to consume the whole session flow control window.
-        CountDownLatch resetLatch = new CountDownLatch(1);
-        MetaData.Request request = newRequest("GET", HttpFields.EMPTY);
-        HeadersFrame frame = new HeadersFrame(request, null, false);
-        FuturePromise<Stream> promise = new FuturePromise<>();
-        client.newStream(frame, promise, new Stream.Listener()
-        {
-            @Override
-            public void onReset(Stream stream, ResetFrame frame, Callback callback)
-            {
-                callback.succeeded();
-                resetLatch.countDown();
-            }
-        });
-        Stream stream = promise.get(5, TimeUnit.SECONDS);
-        ByteBuffer data = ByteBuffer.allocate(((HTTP2Session)client).updateSendWindow(0));
-        stream.data(new DataFrame(stream.getId(), data, true), Callback.NOOP);
+            // Wait for WINDOW_UPDATEs to be processed by the client.
+            sleep(1000);
 
-        assertTrue(resetLatch.await(2 * idleTimeout, TimeUnit.MILLISECONDS));
+            assertThat(((HTTP2Session)client).updateSendWindow(0), Matchers.greaterThan(0));
 
-        // Wait for WINDOW_UPDATEs to be processed by the client.
-        sleep(1000);
-
-        assertThat(((HTTP2Session)client).updateSendWindow(0), Matchers.greaterThan(0));
+            // The server must be stopped before StacklessLogging is closed.
+            LifeCycle.stop(server);
```

Comment:
This feels really wrong.

Reply:
It's for the same reason as for your other comment: since `@AfterEach` can execute before the handler returns, `StacklessLogging` may have re-enabled stack trace logging.